homeassistant/components/group/light.py

```diff
@@ -172,8 +183,32 @@ def should_poll(self) -> bool:
         if ATTR_FLASH in kwargs:
             data[ATTR_FLASH] = kwargs[ATTR_FLASH]
 
-        await self.hass.services.async_call(
-            light.DOMAIN, light.SERVICE_TURN_ON, data, blocking=True)
+        if emulate_color_temp_entity_ids:
```

Comment:
Please use a guard clause:

```python
if not emulate:
    await X
    return

data = 
```

Reply:
I'm not sure I full understand this. The variable is defined early in the method, what are we guarding against here? I have updated the conditional to use a length test instead.

Reply:
Since the "else" code block is only 2 lines long, by turning the if-check around and then returning, you do not need an "else". You then save on indentation and get more readable and maintainable code .

Reply:
Got it, makes sense, fixed.